### PR TITLE
Obfs4 for macOS

### PR DESCRIPTION
**Pull request by codyleblanc**
Includes changes from the obfs4-android branch.

Changes
* Add build instructions for macOS
* Change version name output to be generic (GhostBear instead of obfs4-android)

### CODE REVIEW

File: obfs4proxy/termmon.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ import (
 	"gitlab.com/yawning/obfs4.git/common/log"
 )
 
-var termMonitorOSInit func(*termMonitor) error
+//var termMonitorOSInit func(*termMonitor) error
```
Comment (Abushawish) [inline obfs4proxy/termmon.go]: Is this commented out var intentional?

File: obfs4proxy/termmon.go [hunk 2]
```diff
@@ -120,14 +120,14 @@ func newTermMonitor() (m *termMonitor) {
...
 		//    parent dies before the monitor is initialized.
-		if termMonitorOSInit != nil {
-			// Errors here are non-fatal, since it might still be
-			// possible to fall back to a generic implementation.
-			if err := termMonitorOSInit(m); err == nil {
-				return
-			}
-		}
-		if runtime.GOOS != "windows" {
+		//if termMonitorOSInit != nil {
```
Comment (Abushawish) [inline obfs4proxy/termmon.go]: Is this intentional?

**Review by Abushawish — APPROVED** [review]
Left two comments about commented out code

File: obfs4proxy/termmon.go (hunk 2 above)
Comment (codyleblanc) [reply obfs4proxy/termmon.go]: I wish I could tell you, it looks intentional to me to support running by itself on android? This is from the android work before. I can remove the commented code though to make it cleaner. Thoughts? @Abushawish

Comment (Abushawish) [reply obfs4proxy/termmon.go]: Oh didn't realize it was there before! Probably predates my work. Let's remove it and if we ever need it we can just look back at the commit. Thanks!

**Review by seankelman — APPROVED** [review]

**Pull request merged**